### PR TITLE
Fix config source gen binding value types to null configuration values

### LINKED ISSUE #103859: InvalidOperationException for nullable int in ConfigurationBinder source generator generated code
### Description

The ConfigurationBinder source generator in the AOT template generates code that doesn’t consider the use of nullable value types like `int`, `long`, etc. As a result, binding fails with an `InvalidOperationException` because it tries to parse an empty string as an `int` in generated code.

`Microsoft.Extensions.Configuration.Binder.SourceGeneration` version `8.0.10.26715`

### Reproduction Steps

Configuration class:

```cs
public class LimitSetup
{
    public int? MaxCount { get; set; }
}
```

Binding in Program.cs (ASP web api AOT template):
```cs
builder.Services
    .AddOptions<LimitSetup>()
    .BindConfiguration("LimitSetup");
```

In _appsettings.json_:
```json
"LimitSetup": {
  "MaxCount": null
}
```

csproj file:
```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <InvariantGlobalization>true</InvariantGlobalization>
    <PublishAot>true</PublishAot>
  </PropertyGroup>

</Project>
```

### Expected behavior

In options `LimitSetup.MaxCount` property is `null` value.

### Actual behavior

I get an exception when binding:

```
System.InvalidOperationException: 'Failed to convert configuration value at 'LimitSetup:MaxCount' to type 'System.Int32'.'
```

ConfigurationBinder source generator generate this code:

```cs
if (configuration["MaxCount"] is string value1)
{
    instance.MaxCount = ParseInt(value1, () => configuration.GetSection("MaxCount").Path);
}
```

Value `value1` is empty string if `MaxCount` is `null` in appsettings.json.

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

.NET 8 - ASP.NET Core Web API (Native AOT) template

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/103859

- https://github.com/dotnet/runtime/issues/103859

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs [hunk 1]
```diff
@@ -1544,6 +1544,59 @@ public void CanBindOnParametersAndProperties_RecordWithArrayConstructorParameter
...
 
+        /// <summary>
+        /// Test binding a value parsable types to null configuration values.
+        /// The test ensure the binding will succeed without exceptions and the value will be null (not set).
+        /// </summary>
+        [Fact]
+        public void BindToKnownParsableTypesWithNullValueTest()
+        {
+            string jsonConfig = @"{
+                            ""ParsableValuesClass"": {
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs:1555]: Should we also test the empty string case?

**Review by ericstj — APPROVED** [review]
Nice minimal change (just the large set of baselines).  I had a minor test suggestion.

File: src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs]: good idea. added!

**Pull request merged**